src/app/pages/home/home.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component } from '@angular/core';
+import { BackendService } from '../../services/backend.service';
+import { Router } from '@angular/router';
+
+@Component({
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss']
+})
+export class HomeComponent {
+  searchTerm: string = '';
+  contacts: Array<Object> = [];
+  noContactsFound: boolean = false;
+
+  constructor(private backend: BackendService) {}
+
+  searchDb() {
+    if (this.searchTerm.length === 0) {
+      return (this.contacts = []);
+    }
+
+    this.backend
+      .search(this.searchTerm)
+      .then((contacts: Array<Object>) => {
+        this.contacts = contacts;
+      })
+      .catch(err => {
+        this.contacts = [];
+      });
+  }
+
+  checkSearchResults() {
+    if (this.contacts.length === 0 && this.searchTerm.length !== 0) {
```

Comment:
Instead of having this in another function, this would probably do better in the .then of the call above. Then when you check the length of `this.contacts` you guarantee know it's after you've gotten the newest data set from the backend server. The way that the function is currently written, you don't know for when the `checkSerachResults` function is firing.